FileStorageLevels/ReadLeveledStorage.cpp

```diff
@@ -0,0 +1,309 @@
+#include "Storage.h"
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/serialization/set.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/uuid/uuid_serialize.hpp>
+#include <chrono>
+#include <cstring>
+#include <fcntl.h>
+#include <fstream>
+#include <iostream>
+#include <unistd.h>
+
+namespace fs = std::filesystem;
+using namespace std::chrono_literals;
+
+ReadLeveledStorage::ReadLeveledStorage(const Storage& first_, const Storage& second_)
+{
+    extfile_name = first_.path;
+    extfile_name.replace_extension("extents");
+
+    if(!fs::exists(first_.path) || !loadExtentFile() || signature != boost::lexical_cast<boost::uuids::uuid>(sig_str))
+    {
+        // TODO: log problem
+        std::error_code ec;
+        fs::remove(extfile_name, ec);
+        fs::remove(first_.path, ec);
+        extents.clear();
+        extents.emplace(Extent{.offset = 0, .length = 0});
+        is_fully_primed = false;
+        primed_file_size = 0;
+        signature = boost::lexical_cast<boost::uuids::uuid>(sig_str);
+    }
+    file_size = fs::file_size(second_.path);
```

Comment:
if file_size == 0 then is_fully_primed = true ?

Reply:
whats wrong with it? your `primary` is bound to `secondary` with size 0, so, the primary is fully primed, no?

Reply:
Yes, just asked if in this case we should set is_fully_primed to true here in the ctor?

Reply:
Good point!